README.md

```diff
@@ -80,6 +80,21 @@ extensions:
 
 https://github.com/phylum-dev/cli/tree/main/extensions
 
+### musl support
```

Comment:
I still think the heading here could be improved. This section isn't really talking about `musl support` because we can still build against musl libc as long as we dynamically link.

I think a heading like `libc version` or `libc requirement` would be more accurate.